crates/sui-framework/packages/dwallet-system/sources/eth_dwallet.move

```diff
@@ -0,0 +1,50 @@
+/*
+ * eth_dwallet module provides functionality for managing Ethereum dWallets within the dwallet_system.
+ * It includes structures to represent dWallet capabilities, functions to create and manage Ethereum dWallets,
+ * and native functions to verify message proofs.
+*/
+
+module dwallet_system::eth_dwallet {
+    use dwallet::object::{Self, UID};
+    use dwallet::transfer;
+    use dwallet::tx_context::TxContext;
+    use dwallet_system::dwallet;
+    use dwallet_system::dwallet::{DWalletCap, MessageApproval};
+
+    const EInvalidStateProof: u64 = 1;
+
+    /// Holds the DWalletCap along with Ethereum-specific information.
+    struct EthereumDWalletCap has key {
+        id: UID,
+        dwallet_cap: DWalletCap,
+    }
+
+    /// Creates an Ethereum dWallet capability object by wrapping an existing DWalletCap.
+    public entry fun create_eth_dwallet_cap(
+        dwallet_cap: DWalletCap,
+        ctx: &mut TxContext
+    ) {
+        let eth_dwallet_cap = EthereumDWalletCap {
+            id: object::new(ctx),
+            dwallet_cap,
+        };
+        transfer::share_object(eth_dwallet_cap);
+    }
+
```

Comment:
Do we also want a function that creates a dWallet that is already tied to a policy? @omersadika